File: java/src/estruturas/repetidas/SomaImpares.java
package estruturas.repetidas;

import java.util.Scanner;

/**
 * Problema "soma_impares"
 * Leia 2 valores inteiros X e Y (em qualquer ordem).
 * A seguir, calcule e mostre a soma dos números impares
 * entre eles.
 */
public class SomaImpares {

  /**
   * The entry point of application.
   *
   * @param args the input arguments
   */
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    System.out.println("Digite dois números inteiros ('x' e 'y'): ");
    int x = sc.nextInt();
    int y = sc.nextInt();

    if (x > y) {
      int temp = x;
      x = y;
      y = temp;
    }

    int summation = 0;
    for (int i = x + 1; i < y; i++) {
      if (i % 2 == 1 || i % 2 == -1) {
        summation += i;
      }
    }

    System.out.printf("%nSOMA DOS ÍMPARES = %d%n", summation);

    sc.close();
  }
}

File: python/src/estruturas/repetitivas/soma_impares.py
""" Problema "soma_impares"
 * Leia 2 valores inteiros X e Y (em qualquer ordem).
 * A seguir, calcule e mostre a soma dos números impares entre eles.
"""


def soma_impares() -> None:
    print("Digite dois números inteiros ('x' e 'y'): ")

    x = int(input())
    y = int(input())

    if x > y:
        x, y = y, x

    summation = 0

    for i in range(x + 1, y):
        if i % 2 == 1:
            summation += i

    print(f"SOMA DOS ÍMPARES = {summation}")


if __name__ == "__main__":
    soma_impares()

